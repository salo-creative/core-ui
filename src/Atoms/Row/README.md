# Row

Row is a pure styled component render. Renders a standard `<div>` with themed styling

-------

## Usage

Install

```javascript
yarn add @salo/core-ui
```

Include

```javascript
import Row from '@salo/core-ui/Atoms/Row';
```

Implement as follows

```javascript
<Row>
  // My content
</Row>
```

### For full prop types and usage see storybook info/knobs