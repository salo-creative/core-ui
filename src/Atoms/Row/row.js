import PropTypes from 'prop-types';
import styled from 'styled-components';

const Row = styled.div`
  display: flex;
  width: 100%;
  align-items: ${ ({ alignItems }) => alignItems };
  flex-direction: ${ ({ flexDirection }) => flexDirection };
  flex-wrap: ${ ({ flexWrap }) => flexWrap };
  flex: ${ ({ flex }) => flex };
  justify-content: ${ ({ justifyContent }) => justifyContent };
  padding: ${ ({ padding }) => padding };
`;

Row.defaultProps = {
  alignItems: 'flex-start',
  flex: '0 1 auto',
  flexDirection: 'row',
  flexWrap: 'wrap',
  justifyContent: 'flex-start',
  padding: '0'
};

Row.propTypes = {
  alignItems: PropTypes.oneOf(['flex-start', 'center', 'flex-end', 'stretch', 'unset']),
  flex: PropTypes.string,
  flexDirection: PropTypes.oneOf(['row', 'row-reverse', 'column', 'column-reverse']),
  flexWrap: PropTypes.oneOf(['wrap', 'no-wrap', 'wrap-reverse']),
  justifyContent: PropTypes.oneOf(['space-between', 'center', 'flex-start', 'flex-end']),
  padding: PropTypes.string
};

Row.displayName = 'Row';

export default Row;