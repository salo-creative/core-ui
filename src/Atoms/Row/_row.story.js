import React from 'react';
import { storiesOf } from '@storybook/react';

// load tests
import { withTests } from '@storybook/addon-jest';
import { withKnobs, text, select } from '@storybook/addon-knobs';
import results from '../../../.storybook/jest-test-results.json';

// FEATURED COMPONENT
import { Row } from '../../index';

// README //
import README from './README.md';

// Start of story logic
const stories = storiesOf('Atoms | Grid/Row', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests({ results }));
stories.addParameters({ jest: ['row'] });

stories.add(
  'Basic',
  (() => {
    const alignItems = select('Align items', {
      'Flex start': 'flex-start',
      'Flex end': 'flex-end',
      'Center': 'center',
      'Stretch': 'stretch',
      'Unset': 'unset'
    }, 'flex-start');
    const justifyContent = select('Justify content', {
      'Space between': 'space-between',
      'Center': 'center',
      'Flex start': 'flex-start',
      'Flex end': 'flex-end'
    }, 'flex-start');
    const flexWrap = select('Flex wrap', {
      'Wrap': 'wrap',
      'No wrap': 'no-wrap',
      'Wrap reverse': 'wrap-reverse'
    }, 'flex-start');
    const padding = text('Padding', '0 0 0 0');
    return (
      <Row
        alignItems={ alignItems }
        justifyContent={ justifyContent }
        flexWrap={ flexWrap }
        padding={ padding }
        style={ { background: '#eaeaea' } }
      >
        A lovely Row
      </Row>
    );
  }), { notes: README }
);