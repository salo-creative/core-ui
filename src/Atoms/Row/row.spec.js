import React from 'react';
import 'jest-styled-components';

// HELPERS
import { renderWithTheme } from '../../../test';

// COMPONENT
import Row from './index';

test('Render default Row and styles', async () => {
  // Render
  const { getByTestId, container } = renderWithTheme(
    <Row data-testid='test-row'>A tasty treat</Row>
  );
  const el = getByTestId('test-row');
  // Assert
  expect(el).toHaveTextContent('A tasty treat');
  expect(container.firstChild).toMatchSnapshot();
});

test('Render Row and styles with core props', async () => {
  // Render
  const { getByTestId, container } = renderWithTheme(
    <Row
      padding='20px'
      flexDirection='column'
      alignItems='center'
      justifyContent='space-between'
      flexWrap='no-wrap'
      data-testid='test-row'
    >
      A tasty treat
    </Row>
  );
  const el = getByTestId('test-row');
  // Assert
  expect(el).toHaveTextContent('A tasty treat');
  expect(el).toHaveStyleRule('padding', '20px');
  expect(el).toHaveStyleRule('flex-direction', 'column');
  expect(el).toHaveStyleRule('align-items', 'center');
  expect(el).toHaveStyleRule('justify-content', 'space-between');
  expect(el).toHaveStyleRule('flex-wrap', 'no-wrap');
  expect(container.firstChild).toMatchSnapshot();
});