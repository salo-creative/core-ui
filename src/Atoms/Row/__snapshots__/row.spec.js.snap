// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Row and styles with core props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 20px;
}

<div
  class="c0"
  data-testid="test-row"
>
  A tasty treat
</div>
`;

exports[`Render default Row and styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 0;
}

<div
  class="c0"
  data-testid="test-row"
>
  A tasty treat
</div>
`;
